data_steward/analytics/cdr_ops/combined.py

```diff
@@ -0,0 +1,479 @@
+
+ STRING_AGG(
+     '(SELECT DISTINCT '
+      || '"' || table_name || '" AS table_name '
+      || ',t.person_id           AS person_id '
+      || ',m.src_hpo_id          AS hpo_id '
+      || 'FROM `' || table_schema || '.' || table_name || '` t '
+      || 'JOIN `' || table_schema || '.' || table_id || '` m ' 
+      || 'USING (' || table_name ||'_id) '
+      || 'LEFT JOIN consented c '
+      || ' USING (person_id)'
+      || 'WHERE m.src_hpo_id <> "rdr" AND c.person_id IS NULL)'
+   , ' UNION ALL ')
+ FROM `{{DATASET_ID}}.INFORMATION_SCHEMA.COLUMNS` c
+ JOIN `{{DATASET_ID}}.__TABLES__` t
+  ON t.table_id = '_mapping_' || c.table_name
+ WHERE column_name = 'person_id'
+   AND t.row_count > 0
+);
+
+-- PIDs whose last EHR consent response was affirmative --
+CREATE OR REPLACE TEMP TABLE consented AS
+( 
+  WITH 
+
+  -- For each pid, the latest EHR consent record --
+  pid_last_consent_response AS
+  (SELECT DISTINCT 
+     p.person_id
+    ,LAST_VALUE(o.value_source_concept_id)
+      OVER (
+       PARTITION BY o.person_id
+       ORDER BY o.observation_datetime
+       ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
+      ) AS last_value_source_concept_id
+   FROM `{{DATASET_ID}}.person` p
+    JOIN `{{DATASET_ID}}.observation` o
+     ON p.person_id = o.person_id
+   WHERE observation_source_concept_id = {{EHR_CONSENT_PERMISSION_CONCEPT_ID}})
+
+  -- Only store pids where the latest response is YES --
+  SELECT 
+   person_id
+  FROM pid_last_consent_response
+  WHERE 
+      last_value_source_concept_id = {{YES_CONCEPT_ID}}
+);
+
+EXECUTE IMMEDIATE query;
+''')
+query = tpl.render(
+    DATASET_ID=DATASET_ID,
+    EHR_CONSENT_PERMISSION_CONCEPT_ID=EHR_CONSENT_PERMISSION_CONCEPT_ID,
+    YES_CONCEPT_ID=YES_CONCEPT_ID)
+q(query)
+# -
+
+# ## Date and datetime fields should have the same date
+# The date represented by associated `_date` and `_datetime` fields of the same row should be the same. If there any discrepancies, there may be a bug in the pipeline (i.e. `ensure_date_datetime_consistency`). It may also be useful to report discrepancies to EHR Ops.
+#
+# ### Implementation notes
+# For each associated date/timestamp pair in the dataset the script that follows assembles a universal subquery which can be applied to all pertinent tables and whose results can be compiled together into a single result set that provides relevant troubleshooting information. For legibility, a maximum of 10 problem rows is returned for each table.
+#
+# As an example, the subquery for the `observation` table should look something like this:
+#
+# ```sql
+# SELECT
+#  "observation"          AS table_name
+# ,observation_id         AS row_id
+# ,"observation_date"     AS date_field
+# ,observation_date       AS date_value
+# ,"observation_datetime" AS timestamp_field
+# ,observation_datetime   AS timestamp_value
+# FROM `DATASET_ID.observation`
+# WHERE observation_date<>DATE(observation_datetime)
+# LIMIT 10
+# ```
+#
+
+q(f'''
```

Comment:
this style seems to depart from the rest of the style of this script.  By that, I think you've been defining the query string and then calling `q()` with the defined value.  Not a big thing, but seems to depart from the convention you started with.